src/Features/CSharp/Portable/Copilot/CSharpImplementNotImplementedExceptionFixProvider.cs

```diff
@@ -0,0 +1,175 @@
+            FindReferencesSearchOptions.Default, cancellationToken).ConfigureAwait(false);
+
+        return progress.GetReferencedSymbols();
+    }
+
+    protected override async Task FixAllAsync(
+        Document document, ImmutableArray<Diagnostic> diagnostics,
+        SyntaxEditor editor, CancellationToken cancellationToken)
+    {
+        foreach (var diagnostic in diagnostics)
+        {
+            await FixOneAsync(editor, document, diagnostic, cancellationToken).ConfigureAwait(false);
+        }
+
+        var changedRoot = editor.GetChangedRoot();
+
+        // Get the language specific rule for formatting this construct and call into the
+        // formatter to explicitly format things. Note: all we will format is the new
+        // node as that's the only node that has the appropriate annotation on it.
+        var formattingOptions = await document.GetSyntaxFormattingOptionsAsync(cancellationToken).ConfigureAwait(false);
+        var rules = ImmutableArray.Create(MultiLineConditionalExpressionFormattingRule.Instance);
+        var spansToFormat = FormattingExtensions.GetAnnotatedSpans(changedRoot, new());
+
+        var formattedRoot = CSharpSyntaxFormatting.Instance.GetFormattingResult(changedRoot, spansToFormat, formattingOptions, rules, cancellationToken).GetFormattedRoot(cancellationToken);
+        changedRoot = formattedRoot;
+
+        editor.ReplaceNode(editor.OriginalRoot, changedRoot);
+    }
+
+    private static async Task FixOneAsync(
+        SyntaxEditor editor, Document document, Diagnostic diagnostic, CancellationToken cancellationToken)
+    {
+        var throwNode = diagnostic.Location.FindNode(getInnermostNodeForTie: true, cancellationToken);
+        var methodOrProperty = throwNode.FirstAncestorOrSelf<MemberDeclarationSyntax>();
+        Contract.ThrowIfNull(methodOrProperty);
+        Contract.ThrowIfFalse(methodOrProperty is BasePropertyDeclarationSyntax or BaseMethodDeclarationSyntax);
+        var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+        var memberSymbol = semanticModel.GetRequiredDeclaredSymbol(methodOrProperty, cancellationToken);
+        var references = await FindReferencesAsync(document, memberSymbol, cancellationToken).ConfigureAwait(false);
+        var copilotService = document.GetRequiredLanguageService<ICopilotCodeAnalysisService>();
+
+        var (implementationSuggestion, isQuotaExceeded) = await copilotService.ImplementNotImplementedExceptionAsync(
+            document, throwNode.Span, methodOrProperty, memberSymbol, semanticModel, references, cancellationToken).ConfigureAwait(false);
+
+        MemberDeclarationSyntax replacement;
+        if (isQuotaExceeded)
+        {
+            replacement = AddCommentToMember(methodOrProperty, CSharpFeaturesResources.Error_colon_Quota_exceeded);
+        }
+        else if (implementationSuggestion == null
```

Comment:
unclear why we're using a dictionary for this, instead of justa  string.

Reply:
This dictionary could return a message containing summary of implementation, and for the case when implementation fails to parse, I would be able to paste that summary in the comment.

I would've preferred dictionary but sure, I'll just return string. In any case, we can always make it dictionary again.